src/ddbToEs/ddbToEsHelper.ts

```diff
@@ -42,57 +41,88 @@ export default class DdbToEsHelper {
-                                    type: 'keyword',
-                                    index: true,
+            const promises: any[] = [];
+            Array.from(indicesToCreate).forEach((index: string) => {
+                const alias = this.generateAlias(index);
+                // Only create index when we also need to create an alias
+                if (aliasesToCreate.has(alias)) {
+                    aliasesToCreate.delete(alias);
+                    logger.info(`create index ${index} & alias ${alias}`);
+                    const params = {
+                        index,
+                        body: {
+                            mappings: {
+                                properties: {
+                                    id: {
+                                        type: 'keyword',
+                                        index: true,
+                                    },
+                                    resourceType: {
+                                        type: 'keyword',
+                                        index: true,
+                                    },
+                                    _references: {
+                                        type: 'keyword',
+                                        index: true,
+                                    },
+                                    documentStatus: {
+                                        type: 'keyword',
+                                        index: true,
+                                    },
                                 },
                             },
+                            aliases: { [alias]: {} },
                         },
-                        aliases: { [`${indexName}-alias`]: {} },
-                    },
-                };
-                await this.ElasticSearch.indices.create(params);
-            } else {
-                const indexAliasExistResponse = await this.ElasticSearch.indices.existsAlias({
-                    index: indexName,
-                    name: `${indexName}-alias`,
-                });
-                logger.debug(indexAliasExistResponse);
-                if (!indexAliasExistResponse.body) {
-                    // Create Alias
-                    logger.debug(`create alias ${indexName}-alias`);
-                    await this.ElasticSearch.indices.putAlias({
-                        index: indexName,
-                        name: `${indexName}-alias`,
-                    });
+                    };
+                    promises.push(this.ElasticSearch.indices.create(params));
                 }
-            }
+            });
+
+            Array.from(aliasesToCreate).forEach((alias: string) => {
+                // Create Alias
+                logger.info(`create alias ${alias}`);
+                promises.push(
+                    this.ElasticSearch.indices.putAlias({
+                        index: alias.substring(0, alias.length - 6),
```

Comment:
can we get of this substring operation as well?

Reply:
Sadly no but we can move it into a method -- let me do that

Reply:
I think that it can be done, but it requires some additional refactoring. It seems that keeping a Set of aliases disconnected from the corresponding index names makes this harder.

Maybe it was better to start off with a data structure like:
```
[
 {
  resourceType: string,
  index: string,
  alias: string,
 },
...
]
```
To explicitly keep the relations between index and alias.

There's also a feature request from Abacus to use custom alias names: https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/issues/96, that would make it harder it harder to derive an index from the alias name with string manipulation(custom alias can be anything)

I think that to accomodate the feature request and the multi-tenancy aliases we'll need to eventually do this kind to refactoring, but the current approach works fine for now.

Reply:
Good call out, makes sense